pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -178,10 +179,12 @@ func defaultPredicates() sets.String {
 }
 
 // ApplyFeatureGates applies algorithm by feature gates.
-func ApplyFeatureGates() {
+func ApplyFeatureGates() []api.FeatureDependency {
+	featureDependency := []api.FeatureDependency{}
```

Comment:
If it's required to search by featureGate name, I prefer a map (or DAG) over slice.

But I looked through the code, it seems it's just iterating the `[]FeatureDependency` using `range()` in ValidatePolicy(), so this looks fine to me - don't do early-optimization :grin: